src/app/shared/text/component.tsx

```diff
@@ -1,52 +1,13 @@
 import React, { ReactElement } from 'react';
-import { Text, TextProps } from 'react-native';
-import { createStyles, variables } from '@styles';
-import { TextTheme } from './enums';
+import { TextProps, Text } from 'react-native-ui-lib';
 
-interface Props extends TextProps {
-  children?: React.ReactNode;
-  theme?: TextTheme;
-}
-
-export function AppText({ style: elementStyle = {}, theme, children, ...restProps }: Props): ReactElement {
+export function AppText({ children, ...restProps }: TextProps): ReactElement {
```

Comment:
As mentioned, I'd avoid usage the UILib modifiers, so we will have here something like:
```tsx
import React, { ReactElement } from 'react';
import { ColorsModifiers, MarginModifiers, Text, TextProps, TypographyModifiers } from 'react-native-ui-lib';
import { colors, typographies } from '@styles';

export type AppTextProps = Omit<TextProps, keyof TypographyModifiers | keyof ColorsModifiers | keyof MarginModifiers> & {
  variant?: keyof typeof typographies,
  color?: keyof typeof colors
}

export function AppText({ children, color = 'white', variant = 'small', style, ...restProps }: AppTextProps): ReactElement {
  return (
    <Text
      style={[typographies[variant], { color: colors[color] }, style]}
      {...restProps}>
      {children}
    </Text>
  );
}
```